CMakeLists.txt

```diff
@@ -340,7 +340,10 @@ if (MFEM_USE_SUNDIALS)
   if (MFEM_USE_HIP)
     list(APPEND SUNDIALS_COMPONENTS NVector_Hip)
   endif()
-  find_package(SUNDIALS REQUIRED ${SUNDIALS_COMPONENTS})
+  # The Core component was added in SUNDIALS v7, so we treat it as optional in
+  # order to support older versions.
+  find_package(SUNDIALS REQUIRED ${SUNDIALS_COMPONENTS}
+    OPTIONAL_COMPONENTS Core)
```

Comment:
Is it possible to have something like
```c++
find_package(SUNDIALS 7 COMPONENTS Core ${SUNDIALS_COMPONENTS})
if (NOT SUNDIALS_FOUND)
  find_package(SUNDIALS 0...6 REQUIRED ${SUNDIALS_COMPONENTS})
endif()
```
where the check for SUNDIALS v7 with the `Core` component soft-fails? This works with the CMake "Config" approach but not with the current MFEM "Module" approach to `find_package`.

Reply:
The suggested code will not work because of the [weird behavior of version ranges](https://github.com/AMReX-Codes/amrex/pull/3835#issuecomment-2015836061) and `find_package`. You will need something like this https://github.com/AMReX-Codes/amrex/blob/development/Tools/CMake/AMReXThirdPartyLibraries.cmake#L154.

Reply:
@cjvogl I overlooked that you were only specifying major versions (we have had this come up multiple times recently but those cases involved minor versions too I believe). If you only specify the major versions, then your suggestion might work but it will preclude a hypothetical SUNDIALS 8 from working even if any API changes do not actually break the MFEM interface to SUNDIALS.  

For what is worth, `core` should always be installed with SUNDIALS 7 or newer. I don't think its possible (w/o someone editing the installed SUNDIALSConfig.cmake file) for `core` to be missing.

Reply:
@balos1 , thanks for the reply: good to know that major version numbering may be immune to the CMake issue noted.

> You could also simply add a check after the find_package call like this for extra assurance

I should note the MFEM's current `find_package` call uses a module specified by MFEM (`FindSUNDIALS.cmake`) and not the config specified by SUNDIALS (`SUNDIALSConfig.cmake`) ... so unfortunately `SUNDIALS_VERSION` is not set with the current `find_package` call.

> For what its worth, core should always be installed with SUNDIALS 7 or newer. 

That is good to know, so at least there wont be a silent failure.

Reply:
Given the scenario I was concerned about can't happen (i.e., that components in `SUNDIALS_COMPONENTS` won't be found if `Core` doesn't exist), I'll mark this as resolved.